src/fields/OpenBoundary.H

```diff
@@ -0,0 +1,194 @@
+#ifndef OPEN_BOUNDARY_H_
+#define OPEN_BOUNDARY_H_
+
+#include <AMReX_AmrCore.H>
+#include <cmath>
+
+/** \brief calculate low integer powers base^exp
+ * \param[in] base base of power
+ */
+template<unsigned int exp> AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+amrex::Real pow (amrex::Real base) {
+    using namespace amrex::literals;
+    if constexpr (exp==0) {
+        return 1._rt;
+    } else if constexpr (exp==1) {
+        return base;
+    } else {
+        return pow<exp-1>(base) * base;
+    }
+    return 0._rt; //shut up compiler
+}
```

Comment:
`std::pow` is indeed much slower than multiplications for integer powers. @atmyers could that go to AMReX?